stdlib/experimental/geo/geo.go

```diff
@@ -71,42 +125,15 @@ func generateGetGridFunc() values.Function {
 			}
 
 			var region s2.Region
-
-			if isObjectArgumentOk(box, boxOk) {
-				minLat, minLatOk := box.Get("minLat")
-				minLon, minLonOk := box.Get("minLon")
-				maxLat, maxLatOk := box.Get("maxLat")
-				maxLon, maxLonOk := box.Get("maxLon")
-
-				if !minLatOk || !minLonOk || !maxLatOk || !maxLonOk {
-					return nil, fmt.Errorf("code %d: invalid box specification - must have minLat, minLon, maxLat, maxLon fields", codes.Invalid)
-				}
-
-				region = getRectRegion(minLat.Float(), minLon.Float(), maxLat.Float(), maxLon.Float())
-			} else if isObjectArgumentOk(circle, circleOk) {
-				lat, latOk := circle.Get("lat")
-				lon, lonOk := circle.Get("lon")
-				radius, radiusOk := circle.Get("radius")
-
-				if !latOk || !lonOk || !radiusOk {
-					return nil, fmt.Errorf("code %d: invalid circle specification - must have lat, lon and radius fields", codes.Invalid)
-				}
-
-				region = getCapRegion(lat.Float(), lon.Float(), radius.Float())
-			} else if isArrayArgumentOk(polygon, polygonOk) {
-				points := make([]s2.Point, polygon.Len())
-				for i := 0; i < polygon.Len(); i++ {
-					point := polygon.Get(i).Object()
-					lat, latOk := point.Get("lat")
-					lon, lonOk := point.Get("lon")
-
-					if !latOk || !lonOk {
-						return nil, fmt.Errorf("code %d: invalid polygon point specification - must have lat, lon fields", codes.Invalid)
-					}
-					points[i] = s2.PointFromLatLng(s2.LatLngFromDegrees(lat.Float(), lon.Float()))
-				}
-
-				region = getLoopRegion(points)
+			switch v := geom.(type) {
+			case box:
+				region = getS2RectRegion(v)
+			case circle:
+				region = getS2CapRegion(v)
+			case polygon:
+				region = getS2LoopRegion(v)
+			default:
+				return nil, fmt.Errorf("code %d: unsupported point type: %T", codes.Invalid, geom)
```

Comment:
Can you to change this to
```go

import "github.com/influxdata/flux/internal/errors"

...

return nil, errors.Newf(codes.Invalid, "unsupported point type: %T", geom)
```

We need to return an instance of this special struct to ensure that we return a 4XX (`codes.Invalid`) or 5XX (`codes.Internal`) from our HTTP API.

Could I trouble you to change the other errors in the `geo` package similarly as well?

Reply:
Done. Thank you.